packages/expo-cli/src/commands/upload/BaseUploader.ts

```diff
@@ -108,16 +108,23 @@ export default class BaseUploader {
 
   async _downloadBuild(urlOrPath: string): Promise<string> {
     const filename = path.basename(urlOrPath);
-    const destinationPath = `/tmp/${filename}`;
-    if (await fs.pathExists(destinationPath)) {
-      await fs.remove(destinationPath);
+    // Since we may need to rename the destination path,
+    // add everything to a folder which can be nuked to ensure we don't accidentally use an old build with the same name.
+    const destinationFolder = '/tmp/expo-upload/';
+    const destinationPath = path.join(destinationFolder, filename);
+
+    if (await fs.pathExists(destinationFolder)) {
+      await fs.remove(destinationFolder);
     }
+    await fs.ensureDir(destinationFolder);
+
     if (urlOrPath.startsWith('/')) {
```

Comment:
Since we assume the path needs to be absolute (starts with `/`) maybe we should change it to:
```suggestion
    if (path.isAbsolute(urlOrPath)) {
```